scene/resources/visual_shader.cpp

```diff
@@ -1243,9 +1243,9 @@ Error VisualShader::_write_node(Type type, StringBuilder &global_code, StringBui
 			} else if (in_type == VisualShaderNode::PORT_TYPE_BOOLEAN && out_type == VisualShaderNode::PORT_TYPE_SCALAR_INT) {
 				inputs[i] = src_var + " > 0 ? true : false";
```

Comment:
I think nothing wrong if you add the same for this line and above (scalar to boolean conversion).

Reply:
I saw this and left as is because I couldn't think of a way you could append an operation after false and change the functionality the same way you could with 0 but I may be wrong and you could add them for safety

Reply:
although on an unrelated note I'm not sure if thats how floats should translate to Boolean. Maybe > .5 should translate to true. I doubt it sees much usage considering there are comparator nodes which accomplish that much better.

Reply:
Well, if it's really would lead to the bug then it can be reported and fixed later. About this PR - it really fixes the bug when you pass a boolean to the first port of the 'If' node (which intended to be a float scalar - but it should be parsed correctly anyway).